Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -104,7 +113,49 @@ initiateCallToCustomer rideId = do
         SCS.CallStatus
           { id = callStatusId,
             callId = exotelResponse.callId,
-            rideId = rideId,
+            entityId = rideId.getId,
+            dtmfNumberUsed = Nothing,
+            status = exotelResponse.callStatus,
+            conversationDuration = 0,
+            recordingUrl = Nothing,
+            createdAt = now
+          }
+
+getDriverMobileNumber :: (EncFlow m r, CoreMetrics m, CacheFlow m r, EsqDBFlow m r) => (Id Person.Person, Id DM.Merchant) -> Text -> m CallRes
+getDriverMobileNumber (driverId, merchantId) rcNo = do
+  logTagInfo "rcNo is" rcNo
+  vehicleRCs <- RCQuery.findAllDriverLinkedByRC rcNo
+  logTagInfo "vehicleRcs length" (show $ length vehicleRCs)
+  rcActiveAssociation <- DAQuery.findByRcIsRcActive (map (.id) vehicleRCs) >>= fromMaybeM (InvalidRequest "No active Rc found")
+  oldDriver <- PSQuery.findById rcActiveAssociation.driverId >>= fromMaybeM (InvalidRequest "Old Person not found")
```

Comment:
Make a common function for it which will give driverNumber too else throw error.